### PR TITLE
Add a migration doc for ACP library users

### PR DESCRIPTION
**Pull request by yangyansong-adbe**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: docs/migration.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+      Lifecycle.registerExtension();
+      Signal.registerExtension();
+-     Analytics.registerExtension();
+-     Target.registerExtension();
+-     Places.registerExtension();
+-     Campaign.registerExtension();
+      MobileCore.start(new AdobeCallback () {
+          @Override
+          public void call(Object o) {
+            MobileCore.configureWithAppID("<your_environment_id_from_Launch>");
```
Comment (PravinPK) [inline docs/migration.md:73]: nit: Can we remove reference to launch
<your_environment_id_from_datacollectionUI>
<your_environment_id_of_mobile_property>
<your_environment_id_of_tag_property>

whatever that suits

File: docs/migration.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+      ...
+    }
+  }
+}
+```
+
+### iOS
+
+```objectivec
+#import "ACPCore.h"
```
Comment (PravinPK) [inline docs/migration.md:86]: We will be renaming these imports from ACP to AEP/Mobile right?

File: docs/migration.md [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+### Identity
+
+#### appendVisitorInfoForURL
+- ACP (2.x)
+```javascript
+ACPCore.appendVisitorInfoForURL(baseURL?: String): Promise<?string> 
+```
+- [AEP (1.x)](https://github.com/adobe/aepsdk-react-native/tree/main/packages/core#appendvisitorinfoforurl)
+```javascript
+MobileCore.appendVisitorInfoForURL(baseURL?: String): Promise<?string>
```
Comment (cacheung) [inline docs/migration.md:278]: Identity.appendVisitorInfoForURL?
Need to change for all API under Identity.

File: docs/migration.md [hunk 4]
```diff
@@ -0,0 +1,421 @@
...
+```
+
+#### extensionVersion
+- ACP (2.x)
+```javascript
+ACPCore.extensionVersion(): Promise<string>
+```
+- [AEP (1.x)](https://github.com/adobe/aepsdk-react-native/tree/main/packages/core#extensionversion-1)
+```javascript
+MobileCore.extensionVersion(): Promise<string>
```
Comment (cacheung) [inline docs/migration.md:288]: Why extenstionVersion is under Identity?

Comment (yangyansong-adbe) [reply docs/migration.md]: It's updated to ->
```javascript
ACPIdentity.extensionVersion(): Promise<string>
```

File: docs/migration.md (hunk 1 above)
Comment (cacheung) [reply docs/migration.md:73]: Did you change for this one?

**Comment by cacheung** [comment]
Looks good, just two minor comments

File: docs/migration.md (hunk 1 above)
Comment (yangyansong-adbe) [reply docs/migration.md]: Ha, I only updated the iOS code, I will fix this as well.

**Review by cacheung — APPROVED** [review]

**Comment by yangyansong-adbe** [comment]
@emdobrin @swarna04  I have to merge this PR for now, then I can put a migration link on the Gitbook PR. Let me know if there is any comment. I will fix it in a separate PR.

**Pull request merged**